File: Easy/27-remove-element/try-4.cpp
/*
 * Problem #27: Remove Element
 * Difficulty: Easy
 * Submission: Try 4
 * status: Accepted
 * Language: cpp
 * Date: 21/11/2024, 22:06:57
 * Link: https://leetcode.com/problems/remove-element/
 */

class Solution {
public:
    int removeElement(vector<int>& nums, int val) {
        int c=0,ip=0,pt=nums.size()-1;
    
        while(ip<=pt){
            if(nums[ip]==val){
                swap(nums[ip],nums[pt]);
                pt--;
                
                continue;
            }
            ip++;
            c++;
        }
        return c;
    }

};

File: Medium/739-daily-temperatures/try-4.py
/*
 * Problem #739: Daily Temperatures
 * Difficulty: Medium
 * Submission: Try 4
 * status: Accepted
 * Language: python3
 * Date: 22/12/2024, 22:22:04
 * Link: https://leetcode.com/problems/daily-temperatures/
 */

@cache
class Solution:
    def dailyTemperatures(self, temperatures: List[int]) -> List[int]:
        stack = []
        re = []
        
        for i in range(len(temperatures)-1,-1,-1):
            while stack and temperatures[stack[-1]]<=temperatures[i] :
                stack.pop()
            if stack:
                re.append(stack[-1]-i)
            else :
                re.append(0)
            stack.append(i)
        
        return re[::-1]
